widget/embedded/src/utils/wallets.ts

```diff
@@ -199,34 +199,35 @@ export function prepareAccountsForWalletStore(
   return result;
 }
 
-export function getRequiredChains(quote: SelectedQuote | null) {
-  const wallets: string[] = [];
+export function getQuoteWallets(params: {
+  filter: 'all' | 'required';
+  quote: SelectedQuote | null;
+}): string[] {
+  const { filter, quote } = params;
+  const wallets = new Set<string>();
 
   quote?.swaps.forEach((swap) => {
     const currentStepFromBlockchain = swap.from.blockchain;
     const currentStepToBlockchain = swap.to.blockchain;
-    if (!wallets.includes(currentStepFromBlockchain)) {
-      wallets.push(currentStepFromBlockchain);
-    }
-    if (!wallets.includes(currentStepToBlockchain)) {
-      wallets.push(currentStepToBlockchain);
-    }
+    wallets.add(currentStepFromBlockchain);
 
     // Check if internalSwaps array exists
-    if (swap.internalSwaps && Array.isArray(swap.internalSwaps)) {
+    if (swap.internalSwaps) {
```

Comment:
I think we could still have this check.

Reply:
It's redundant. The `internalSwaps` type is a union of null and array. If `swap.internalSwaps` is not falsy, TypeScript recognizes it as an array within the if block.